src/Tests/BusinessLayer/TelemetryReportingTest.cpp

```diff
@@ -0,0 +1,959 @@
+    EXPECT_CALL(*communicationService_, sendData(_, EXPECTED_PACKAGE_LENGTH_SEND_MPPT)).With(Args<0, 1>(packageIdIs(9))).Times(CcsDefines::MPPT_COUNT);
+    EXPECT_CALL(*communicationService_, sendData(_, EXPECTED_PACKAGE_LENGTH_SEND_LIGHTS)).With(Args<0, 1>(packageIdIs(10))).Times(1);
+    telemetryReporting_->sendAll();
+}
+
+/*
+ * Tests if Consistent Overhead Byte Stuffing (COBS) is working as intended
+ */
+TEST_F(TelemetryReportingTest, COBSTest)
+{
+    unsigned char nullArrayInput[] = {0x00};
+    unsigned char nullArrayExpected[] = {0x01, 0x01, 0x00};
+    assertThatCOBScorrect(nullArrayInput, nullArrayExpected, 1);
+    unsigned char input1[] = {0x00, 0x00};
+    unsigned char expected1[] = {0x01, 0x01, 0x01, 0x00};
+    assertThatCOBScorrect(input1, expected1, 2);
+    unsigned char input2[] = {0x11, 0x22, 0x00, 0x33};
+    unsigned char expected2[] = {0x03, 0x11, 0x22, 0x02, 0x33, 0x00};
+    assertThatCOBScorrect(input2, expected2, 4);
+    unsigned char input3[] = {0x11, 0x22, 0x33, 0x44};
+    unsigned char expected3[] = {0x05, 0x11, 0x22, 0x33, 0x44, 0x00};
+    assertThatCOBScorrect(input3, expected3, 4);
+    unsigned char input4[] = {0x11, 0x00, 0x00, 0x00};
+    unsigned char expected4[] = {0x02, 0x11, 0x01, 0x01, 0x01, 0x00};
+    assertThatCOBScorrect(input4, expected4, 4);
+    const unsigned int MAX_COBS_PACKAGE_LENGTH = 254;
+    unsigned char inputAlmostMax[MAX_COBS_PACKAGE_LENGTH - 1];
+    unsigned char expectedAlmostMax[MAX_COBS_PACKAGE_LENGTH + 1];
+
+    for (unsigned char i = 0; i < MAX_COBS_PACKAGE_LENGTH - 1; ++i)
+    {
+        inputAlmostMax[i] = i + 1;
+        expectedAlmostMax[i + 1] = i + 1;
+    }
+
+    expectedAlmostMax[0] = 0xFE;
+    expectedAlmostMax[MAX_COBS_PACKAGE_LENGTH] = 0x00;
+    assertThatCOBScorrect(inputAlmostMax, expectedAlmostMax, MAX_COBS_PACKAGE_LENGTH - 1);
+    unsigned char inputMax[MAX_COBS_PACKAGE_LENGTH];
+    unsigned char expectedMax[MAX_COBS_PACKAGE_LENGTH + 2];
+
+    for (unsigned char i = 0; i < MAX_COBS_PACKAGE_LENGTH; ++i)
+    {
+        inputMax[i] = i + 1;
+        expectedMax[i + 1] = i + 1;
+    }
+
+    expectedMax[0] = 0xFF;
+    expectedMax[MAX_COBS_PACKAGE_LENGTH + 1] = 0x00;
+    assertThatCOBScorrect(inputMax, expectedMax, MAX_COBS_PACKAGE_LENGTH); // TODO this one fails. can probably be fixed by removing line 452 from TelemetryReporting.cpp
+    // TODO !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    // TODO WHAT HAPPENS IF THE Distance to the first zero is more than 254 bytes?? (our packages are smaller...)
+    // TODO !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+}
+
+/*
+ * This test tests if the checksum is correctly appended. It does not test the checksum calculation itself.
+ * (Supposed to work correctly, as copied from already tested code?!)
```

Comment:
Yup, this is true, this file doesn't need to test the CRC Calculator. You can remove this line.